### ISSUE TITLE
Add check for header `Feature-Policy`

### ISSUE THREAD

**Issue by craigfrancis** [mozilla/http-observatory#361]
The `Feature-Policy` header (and `allow` attribute on <iframe>'s) is beginning to get support in Chrome (as of ~60), and hopefully other browsers soon:

https://wicg.github.io/feature-policy/
https://developers.google.com/web/updates/2018/06/feature-policy
https://cs.chromium.org/chromium/src/third_party/blink/renderer/platform/feature_policy/feature_policy.cc?q=GetDefaultFeatureNameMap&sq=package:chromium&dr=CSs&l=138

e.g.

    Feature-Policy:
        accelerometer: 'none';
        ambient-light-sensor: 'none';
        camera: 'none';
        encrypted-media: 'none';
        fullscreen: 'none';
        geolocation: 'none';
        gyroscope: 'none';
        magnetometer: 'none';
        microphone: 'none';
        midi: 'none';
        payment: 'none';
        speaker: 'none';
        sync-xhr: 'self';
        usb: 'none';
        vr: 'none';

**Comment by maximcherny** [comment]
This would be a great addition.

Implementing it, however, would require an agreed set of principles for how evaluation of the relevant values should be done, similarly to `content_security_policy` test, as both follow similar convention for how values can be set.

The `content_security_policy` test (see: https://github.com/mozilla/http-observatory/blob/master/httpobs/scanner/analyzer/headers.py).

Please refer to the code after the `# If we make it this far, we have a policy object` comment.

What would be the most appropriate logic for evaluating the perceived safety of the `Feature-Policy` header?

**Comment by craigfrancis** [comment]
I would go with a simple check to see if the header exists (+5).

And *maybe* an analysis table, listing the directives browsers currently support (something that will change over time), with how the website is using them.

This table could highlight directives the website author hasn't set (possibly because they aren't aware of new ones being added), or note when they have been set to 'none' (very good), 'self' (good), a specific host (probably good), or a wildcard `*` (probably not good, but better than not being set at all).

The only directive I think we could specifically recommend is `document-write`, as that implies an insecure coding style. All the other directives are nice to have (e.g. disable  `geolocation` if you don't need it).

**Comment by mozdevcontrib** [comment]
> I would go with a simple check to see if the header exists (+5).
> 
> And _maybe_ an analysis table, listing the directives browsers currently support (something that will change over time), with how the website is using them.
> 
> This table could highlight directives the website author hasn't set (possibly because they aren't aware of new ones being added), or note when they have been set to 'none' (very good), 'self' (good), a specific host (probably good), or a wildcard `*` (probably not good, but better than not being set at all).
> 
> The only directive I think we could specifically recommend is `document-write`, as that implies an insecure coding style. All the other directives are nice to have (e.g. disable `geolocation` if you don't need it).

Agreed on both counts, however the other "features" as being nice to have I'd have to disagree. Would you like having your mic flipped on without your knowledge? Your cam? Those two alone are utterly creepy, but then let's move on to payments - how about having your credit card drained or even just having a bit taken off the top by devs acting in bad faith? Wake-lock - while I can think of scenarios that using it for malicious ends; they are yet, I believe, though correct me if I'm mistaken, theoretical / academic. But it would drain the battery on laptops and mobile devices fairly quickly - which technically is a malicious act, just not in the traditional coding sense. The same for the vibrate function. Basically, if web apps / sites do not need nor use any "feature" for valid reasons, then the features SHOULD BE disabled from the start.
